lib/authentication/secure_storage/json_credential_manager.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const path = require('path');
+const Logger = require('../../logger');
+const fs = require('fs');
+const os = require('os');
+const Util = require('../../util');
+
+function JsonCredentialManager() {
+
+  this.userinfo = os.userInfo();
+
+  this.getTokenDir = function () {
+    let tokenDir = process.env.SF_TEMPORARY_CREDENTIAL_CACHE_DIR;
+
+    if (!Util.exists(tokenDir)) {
+      tokenDir = os.homedir();
+    } else {
+      Logger.getInstance().info(`The credential cache directory is configured by the user. The token will be saved at ${tokenDir}`);
+    }
+
+    if (!Util.exists(tokenDir)) {
+      Logger.getInstance().error(`Temporary credential cache directory is invalid, and the driver is unable to use the default location(home). 
+      Please assign the enviroment variable value SF_TEMPORARY_CREDENTIAL_CACHE_DIR to enable the default credential manager.`);
+      return null; 
+    }
+    const tokenCacheFile = path.join(tokenDir, 'temporary_credential.json');
+
+    if (fs.existsSync(tokenCacheFile)) {
+      try {
+        fs.accessSync(tokenCacheFile,  fs.constants.R_OK | fs.constants.W_OK);
+      } catch (err) {
+        Logger.getInstance().info(`You do not have permission to the credential file: ${tokenCacheFile}.`);
+        return null;
```

Comment:
should we only log on info level and return null? should we throw the exception?

Reply:
The same idea on above. Please let me know if we need to throw an error in this case or if we just need to disable the credential cache.

Reply:
let's keep one thread for discussing it - I am closing this comment